src/components/v2v3/V2V3Project/ProjectSafeDashboard/juiceboxTransactions/reconfigureFundingCyclesOf/ReconfigurationRichPreview.tsx

```diff
@@ -0,0 +1,101 @@
+import { t, Trans } from '@lingui/macro'
+import { Space } from 'antd'
+import { MinimalCollapse } from 'components/MinimalCollapse'
+import SplitList from 'components/v2v3/shared/SplitList'
+import FundingCycleDetails from 'components/v2v3/V2V3Project/V2V3FundingCycleSection/FundingCycleDetails'
+import { ThemeContext } from 'contexts/themeContext'
+import { V2V3ContractsContext } from 'contexts/v2v3/V2V3ContractsContext'
+import { V2V3ProjectContext } from 'contexts/v2v3/V2V3ProjectContext'
+import { useContext } from 'react'
+import { formatOutgoingSplits } from 'utils/splits'
+import { formatReservedRate, MAX_DISTRIBUTION_LIMIT } from 'utils/v2v3/math'
+import { SafeTransactionComponentProps } from '../../SafeTransaction'
+
+export function ReconfigureRichPreview({
+  transaction,
+}: SafeTransactionComponentProps) {
+  const { contracts } = useContext(V2V3ContractsContext)
+  const { projectOwnerAddress } = useContext(V2V3ProjectContext)
+  const {
+    theme: { colors },
+  } = useContext(ThemeContext)
+
+  if (!contracts) return null
+
+  const decodedData = contracts['JBController'].interface.parseTransaction({
+    data: transaction.data ?? '',
+  }).args
+
+  const distributionLimitCurrency =
+    decodedData._fundAccessConstraints[0].distributionLimitCurrency
+  const distributionLimit =
```

Comment:
I guess we could be more safe here and add some guards:

```suggestion
  const distributionLimit =
      decodedData._fundAccessConstraints?.[0]?.distributionLimit
```

Especially if we keep the `any` type. Can imagine weird situations where this might break otherwise.